scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -372,20 +375,14 @@ abstract class ScanResultsStorage : PackageBasedScanStorage {
         read(pkg, scannerCriteria).toNestedProvenanceScanResult(nestedProvenance)
 
     private fun Result<List<ScanResult>>.toNestedProvenanceScanResult(nestedProvenance: NestedProvenance) =
-        when (this) {
-            is Success -> {
-                result.filter { it.provenance == nestedProvenance.root }
-                    .map { it.toNestedProvenanceScanResult(nestedProvenance) }
-            }
-
-            is Failure -> throw ScanStorageException(error)
-        }
+        map { scanResults ->
+            scanResults.filter { it.provenance == nestedProvenance.root }
+                .map { it.toNestedProvenanceScanResult(nestedProvenance) }
+        }.getOrThrow()
```

Comment:
Is the encapsulated exception always a `ScanStorageException`?

Reply:
I believe so. In my porting, I payed attention to never throw anything else than `ScanStorageException`.